Index: resources/sunspider-standalone-driver.js
===================================================================
--- resources/sunspider-standalone-driver.js	(revision 77121)
+++ resources/sunspider-standalone-driver.js	(working copy)
@@ -23,6 +23,25 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
  */
 
+// To make the benchmark results predictable, we replace Math.random
+// with a 100% deterministic alternative.
+// Avoid using closure for seed for a minimal overhead.
+var random_seed = 49734321; 
+
+Math.random = function() {
+    // Robert Jenkins' 32 bit integer hash function.
+    random_seed = ((random_seed + 0x7ed55d16) + (random_seed << 12))  & 0xffffffff;
+    random_seed = ((random_seed ^ 0xc761c23c) ^ (random_seed >>> 19)) & 0xffffffff;
+    random_seed = ((random_seed + 0x165667b1) + (random_seed << 5))   & 0xffffffff;
+    random_seed = ((random_seed + 0xd3a2646c) ^ (random_seed << 9))   & 0xffffffff;
+    random_seed = ((random_seed + 0xfd7046c5) + (random_seed << 3))   & 0xffffffff;
+    random_seed = ((random_seed ^ 0xb55a4f09) ^ (random_seed >>> 16)) & 0xffffffff;
+    return (random_seed & 0xfffffff) / 0x10000000;
+};
+
+for (var i = 0; i != 50; ++i)
+    Math.random();
+
 var results = new Array();
 
 (function(){
@@ -33,7 +52,6 @@
 
 for (var j = 0; j < tests.length; j++) {
     var testName = suitePath + "/" + tests[j] + ".js";
-    var startTime = new Date;
     if (testName.indexOf('parse-only') >= 0)
         times[j] = checkSyntax(testName);
     else
